### PR TITLE
Add rotation module and functions

### PR DESCRIPTION
**Pull request by ripytide**
As a part of https://github.com/image-rs/image/issues/2238

### CODE REVIEW

**Comment by theotherphil** [comment]
Thanks! These should go alongside the existing rotation functions in the (clunkily named) geometric_transformation module.

**Comment by ripytide** [comment]
I've moved them across :+1:

**Comment by ripytide** [comment]
Do you think we should move the `flip` functions to `geometric_transformations` as well since they are like reflections?

**Comment by theotherphil** [comment]
Let’s leave them for now and bike shed the module structure with whatever functions we’ve got before the next release :-)

File: src/geometric_transformations.rs [hunk 1]
```diff
@@ -317,6 +317,146 @@ where
...
+///
+/// let rotated = rotate90(&image);
+///
+/// assert_eq!(rotated, gray_image!(
+///     0, 0, 3, 1;
+///     0, 0, 4, 2;
+///     0, 0, 0, 0;
+///     0, 0, 0, 0));
+/// ```
+pub fn rotate90<P>(image: &Image<P>) -> Image<P>
```
Comment (theotherphil) [inline src/geometric_transformations.rs:341]: I don’t feel strongly either way but it’s a bit unfortunate to use angles in degrees in the names of these functions when angles everywhere else in this file (/crate) are in radians. Can’t think of better names off the top of my head though.

Comment (ripytide) [reply src/geometric_transformations.rs:341]: As some prior art, opencv just has a single `rotate()` function which takes an enum that looks like `ROTATE_90_CLOCKWISE`

Comment (ripytide) [reply src/geometric_transformations.rs:341]: Maybe we should do the same and use a single function with an enum, but I don't feel particularly strongly either way.

Comment (theotherphil) [reply src/geometric_transformations.rs:341]: I wondered the same thing. I’ll just merge as is for now. Thanks.

**Pull request merged**